### ISSUE TITLE
Reorganisation & fixes for the next release

### ISSUE THREAD

**Issue by garyb** [purescript-contrib/purescript-dom#93]
This won't necessarily happen in the near future, since it will break a lot of stuff, but there are a bunch of things that should be updated / made consistent:

- [ ] Use a consistent `URL` newtype throughout
- [ ] Rename `DOM.Websocket` to `DOM.WebSocket`
- [ ] `MessageEvent` is under websockets, but is more generally applicable

(& will probably extend over time)

**Comment by natefaubion** [comment]
I think there are a few APIs that should be moved out into their own library within `purescript-web`.

* DOM.XHR
* DOM.WebStorage
* DOM.WebSocket
* DOM.File

**Comment by chexxor** [comment]
I've got `DOM.HTML.Navigator.Geolocation` in my [chexxor/purescript-dom-geolocation - use-aff branch - DOM/HTML/Navigator/Geolocation.purs](https://github.com/chexxor/purescript-dom-geolocation/blob/use-aff/src/DOM/HTML/Navigator/Geolocation.purs) Idk if it needs a rename.
Would that belong here as "purescript-web/purescript-dom-geolocation"?

**Comment by gabejohnson** [comment]
I'm planning to move websocket out to https://github.com/purescript-web/purescript-web-socket. The same with a bunch of other Web APIs. I was just waiting on the removal of effect rows from `Eff`.

My instinct is to have separate PRs for `Eff` -> `Effect` and then one for each new lib.

Thoughts?

**Comment by garyb** [comment]
So make an `-> Effect` PR in here, and then break stuff out after you mean? I guess that makes sense, in case there are any changes during that that need thinking about.

I'll perhaps merge into a new branch on here though, until we're ready to release the new "web / dom ecosystem" based on `Effect` though, rather than having it in master in the mean time. Sound good?

**Comment by gabejohnson** [comment]
Sounds good. I'll submit the PR. I can set it to merge with `master`, or if you want to make the new branch first I'll wait.

**Comment by garyb** [comment]
I've pushed an `effect` branch here now :)
